torch/nn/functional.py

```diff
@@ -547,6 +547,22 @@ def nll_loss(input, target, weight=None, size_average=True):
     return f(input, target)
 
 
+def poisson_nll_loss(log_input, target, size_average=True):
+    r"""Poisson negative log likelihood loss.
+
+    Drops :math:`log(target!)` term. See :class:`~torch.nn.PoissonNLLLoss` for details.
+
+    Args:
+        log_input: :math:`log(input)`
+        target: :math:`target \sim Pois(input)`
+    """
+    loss = torch.exp(log_input) - target * log_input
```

Comment:
Is there a reason why you expect `log(input)` instead of `input` as the input? Couldn't you do instead
```python
loss = input - target * torch.log(input)
```
?
Also, what's the relationship with `nll_loss`? Do you have a paper that proposes this loss?
Because if it is somewhat similar to `nll_loss`, you'd need to consider that the loss only takes into account the value of the input at a specified target index.